tests/ftqc/test_cond_meas.py

```diff
@@ -0,0 +1,249 @@
+            m = qml.measure(0)
+            cond_meas(m, measure_x, inp)(0)
+
+    @pytest.mark.parametrize("attribute, inp", [("reset", (True, False)), ("postselect", (0, 1))])
+    def test_mismatched_settings_raises_error(self, attribute, inp):
+        """Test that a mismatch between the measurement settings `reset` and `postselect`
+        on the two measurements raises an error"""
+
+        input1 = {attribute: inp[0]}
+        input2 = {attribute: inp[1]}
+
+        with pytest.raises(
+            ValueError,
+            match="behaviour must be consistent for both branches",
+        ):
+            m = qml.measure(0)
+            cond_meas(m, partial(measure_y, **input1), partial(measure_x, **input2))(0)
+
+    def test_mismatched_wires_raises_error(self):
+
+        with pytest.raises(
+            ValueError,
+            match="behaviour must be consistent for both branches",
+        ):
+            m = qml.measure(0)
+            cond_meas(m, partial(measure_y, wires=0), partial(measure_x, wires=1))()
+
+    @pytest.mark.jax
+    def test_program_capture_raises_error(self):
+        """Test that a clear error is raised when executing with program capture"""
+        try:
+            qml.capture.enable()
+
+            with pytest.raises(NotImplementedError, match="not compatible with program capture"):
+                m = qml.measure(0)
+                cond_meas(m, measure_x, measure_y)(0)
+        finally:
+            qml.capture.disable()
+
+
+class TestWorkflows:
+
+    @pytest.mark.parametrize("mcm_method, shots", [("tree-traversal", None), ("one-shot", 10000)])
+    def test_execution_in_cond(self, mcm_method, shots):
+        """Test that we can execute a QNode with a ParametricMidMeasureMP applied in a conditional,
+        and produce an accurate result"""
+
+        dev = qml.device("default.qubit", shots=shots)
+
+        @diagonalize_mcms
+        @qml.qnode(dev, mcm_method=mcm_method)
+        def circ():
+            qml.RX(np.pi, 0)
+            m = qml.measure(0)  # always 1
+
+            qml.RX(2.345, 1)
+            cond_meas(m == 0, measure_x, measure_y)(1)  # always measure_y
+            return qml.expval(qml.Z(1))
+
+        if shots:
+            # the result is on the order of 1 (-0.7), and an uncertainty ~1.5-2 orders of magnitude
+            # smaller than the result is sufficiently accurate for a shots-based measurement
+            assert np.isclose(circ(), -np.sin(2.345), atol=0.03)
+        else:
+            assert np.isclose(circ(), -np.sin(2.345))
+
+    @pytest.mark.parametrize("mcm_method, shots", [("tree-traversal", None), ("one-shot", 10000)])
+    def test_cascading_conditional_measurements(self, mcm_method, shots):
+        """Test a workflow that feeds measurement values from conditional measurements forward
+        into subsequent measurements and operations applied in `cond_meas` and `cond`"""
+
+        dev = qml.device("default.qubit", shots=shots)
+
+        @diagonalize_mcms
```

Comment:
Should we test that we have incorrect outcomes where this line is excluded, or is it fine to validate that use of it gives the correct outcome?

Reply:
Sure, can't hurt to validate that the test circuit is an example that actually _needs_ the transform to work

Reply:
Added! Without the transform, one fails by returning incorrect results (but forgetting to diagonalize is one of the few places where we generally consider "wrong results" to be user-error and not a bug). The other fails with an error.